trino/sqlalchemy/datatype.py

```diff
@@ -0,0 +1,169 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import re
+from typing import Dict, Iterator, Type, Union
+
+from sqlalchemy import util
+from sqlalchemy.sql import sqltypes
+from sqlalchemy.sql.type_api import TypeEngine
+
+# https://trino.io/docs/current/language/types.html
+_type_map = {
+    # === Boolean ===
+    'boolean': sqltypes.BOOLEAN,
+
+    # === Integer ===
+    'tinyint': sqltypes.SMALLINT,
+    'smallint': sqltypes.SMALLINT,
+    'integer': sqltypes.INTEGER,
+    'bigint': sqltypes.BIGINT,
+
+    # === Floating-point ===
+    'real': sqltypes.FLOAT,
+    'double': sqltypes.FLOAT,
+
+    # === Fixed-precision ===
+    'decimal': sqltypes.DECIMAL,
+
+    # === String ===
+    'varchar': sqltypes.VARCHAR,
+    'char': sqltypes.CHAR,
+    'varbinary': sqltypes.VARBINARY,
+    'json': sqltypes.JSON,
+
+    # === Date and time ===
+    'date': sqltypes.DATE,
+    'time': sqltypes.TIME,
+    'timestamp': sqltypes.TIMESTAMP,
+
+    # 'interval year to month':
+    # 'interval day to second':
+    #
+    # === Structural ===
+    # 'array': ARRAY,
+    # 'map':   MAP
+    # 'row':   ROW
+    #
+    # === Mixed ===
+    # 'ipaddress': IPADDRESS
+    # 'uuid': UUID,
+    # 'hyperloglog': HYPERLOGLOG,
+    # 'p4hyperloglog': P4HYPERLOGLOG,
+    # 'qdigest': QDIGEST,
+    # 'tdigest': TDIGEST,
+}
+
+SQLType = Union[TypeEngine, Type[TypeEngine]]
+
+
+class MAP(TypeEngine):
+    __visit_name__ = "MAP"
+
+    def __init__(self, key_type: SQLType, value_type: SQLType):
+        if isinstance(key_type, type):
+            key_type = key_type()
+        self.key_type: TypeEngine = key_type
+
+        if isinstance(value_type, type):
+            value_type = value_type()
+        self.value_type: TypeEngine = value_type
+
+    @property
+    def python_type(self):
+        return dict
+
+
+class ROW(TypeEngine):
+    __visit_name__ = "ROW"
+
+    def __init__(self, attr_types: Dict[str, SQLType]):
+        for name, attr_type in attr_types.items():
+            if isinstance(attr_type, type):
+                attr_type = attr_type()
+                attr_types[name] = attr_type
+        self.attr_types: Dict[str, TypeEngine] = attr_types
+
+    @property
+    def python_type(self):
+        return dict
+
+
+def split(string: str, delimiter: str = ',',
+          quote: str = '"', escaped_quote: str = r'\"',
+          open_bracket: str = '(', close_bracket: str = ')') -> Iterator[str]:
+    """
+    A split function that is aware of quotes and brackets/parentheses.
```

Comment:
Since you have already added a comment, an example (or a self-explanatory method name) would help even more because the method impl is not very straightforward.

Reply:
It's renamed to `aware_split` to distinguish it from `str.split()`